### PR TITLE
add zstd support

### LINKED ISSUE #23: Add ZSTD compression
Requires libzstd and probably GDAL configuration flags.

### PR DESCRIPTION
**Pull request by vincentsarago**
resolves #23

### CODE REVIEW

File: config.sh [hunk 1]
```diff
@@ -145,6 +145,16 @@ function build_curl {
     touch curl-stamp
 }
 
+function build_zstd {
+    CFLAGS="$CFLAGS -g -O2"
+    CXXFLAGS="$CXXFLAGS -g -O2"
+    if [ -e zstd-stamp ]; then return; fi
+    fetch_unpack https://github.com/facebook/zstd/archive/v${ZSTD_VERSION}.tar.gz
+    (cd zstd-${ZSTD_VERSION}  \
```
Comment (vincentsarago) [inline config.sh:153]: I couldn't test if `fetch_unpack` was really creating a `zstd-${ZSTD_VERSION}` directory 🤞

File: config.sh [hunk 2]
```diff
@@ -145,6 +145,16 @@ function build_curl {
     touch curl-stamp
 }
 
+function build_zstd {
+    CFLAGS="$CFLAGS -g -O2"
+    CXXFLAGS="$CXXFLAGS -g -O2"
```
Comment (vincentsarago) [inline config.sh:150]: this is copied from other builds, might not be necessary!

**Comment by sgillies** [comment]
Thanks @vincentsarago !

**Comment by sgillies** [comment]
@vincentsarago the zstd library doesn't install. Here's the tail of the output from `make install`:

```
creating pkgconfig
sed: invalid option -- E
Usage: sed [OPTION]... {script-only-if-no-other-script} [input-file]...
  -n, --quiet, --silent
                 suppress automatic printing of pattern space
  -e script, --expression=script
                 add the script to the commands to be executed
  -f script-file, --file=script-file
                 add the contents of script-file to the commands to be executed
  -i[SUFFIX], --in-place[=SUFFIX]
                 edit files in place (makes backup if extension supplied)
  -c, --copy
                 use copy instead of rename when shuffling files in -i mode
		 (avoids change of input file ownership)
  -l N, --line-length=N
                 specify the desired line-wrap length for the `l' command
  --posix
                 disable all GNU extensions.
  -r, --regexp-extended
                 use extended regular expressions in the script.
  -s, --separate
                 consider files as separate rather than as a single continuous
                 long stream.
  -u, --unbuffered
                 load minimal amounts of data from the input files and flush
                 the output buffers more often
      --help     display this help and exit
      --version  output version information and exit
If no -e, --expression, -f, or --file option is given, then the first
non-option argument is taken as the sed script to interpret.  All
remaining arguments are names of input files; if no input files are
specified, then the standard input is read.
E-mail bug reports to: bonzini@gnu.org .
Be sure to include the word ``sed'' somewhere in the ``Subject:'' field.
make[1]: *** [libzstd.pc] Error 4
make[1]: Leaving directory `/io/zstd-1.4.7/lib'
make: *** [install] Error 2
```

**Comment by vincentsarago** [comment]
@sgillies not sure what's going on, looking at zstd repo I found https://github.com/facebook/zstd/issues/2215
definitely something I never saw. Is it not installing for all env ?

**Comment by sgillies** [comment]
Tomorrow I'll try `make -j4 PREFIX=$BUILD_PREFIX ZSTD_LEGACY_SUPPORT=0 SED_ERE_OPT="-r"`.

**Comment by sgillies** [comment]
Actually, would need to be `make install SED_ERE_OPT="-r"`

**Pull request merged**